font/src/ft/mod.rs

```diff
@@ -293,19 +283,24 @@ impl FreeTypeRasterizer {
         };
 
         // Reuse the font_key, since changing it can break library users
-        let font_key = self.face_from_pattern(&base_font, font_key).and_then(|pattern| {
-            pattern.map(Ok).unwrap_or_else(|| Err(Error::MissingFont(desc.to_owned())))
-        })?;
+        let font_key = self.face_from_pattern(&primary_font, primary_font_id, font_key).and_then(
+            |pattern| pattern.map(Ok).unwrap_or_else(|| Err(Error::MissingFont(desc.to_owned()))),
+        )?;
```

Comment:
If you want to `?` only once, you should do it like this:

```
        let font_key = self
            .face_from_pattern(&primary_font, primary_font_id, font_key)
            .and_then(|pattern| pattern.ok_or_else(|| Error::MissingFont(desc.to_owned())))?;
```

If we don't care about that, this is another option:

```
        let font_key = self
            .face_from_pattern(&primary_font, primary_font_id, font_key)?
            .ok_or_else(|| Error::MissingFont(desc.to_owned()))?;
```